### PR TITLE
add util.bootstrap

### PR DESCRIPTION
**Pull request by chicoxyzzy**

### CODE REVIEW

**Comment by chicoxyzzy** [comment]
I will add more progress events

File: util.js [hunk 1]
```diff
@@ -627,3 +629,69 @@ export function format (format, ...args) {
 
   return str
 }
+
+// TODO: move to WebWorker?
+export async function bootstrap ({ src, dest, hashDownloaded, hashUnpacked, hashAlgorithm, unpack }) {
+  if (!src || !dest) {
+    throw new Error('src and dest are required')
```
Comment (heapwolf) [inline util.js]: `.src and .dest are required string properties on the object provided to the constructor at the first argument position`

File: bootstrap.js [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+    }
+    return unpacked
+  }
+
+  cleanup () {
+    this.removeAllListeners()
+  }
+}
+
+// TODO: move to WebWorker?
```
Comment (jwerle) [inline bootstrap.js]: probably but we'll need `window._ipc` to tbe there

Comment (chicoxyzzy) [reply bootstrap.js]: hmm. it could cause a performance downgrade

File: bootstrap.js [hunk 3]
```diff
@@ -0,0 +1,142 @@
+import { readFile } from './fs/promises.js'
+import { createWriteStream } from './fs/index.js'
+import { PassThrough } from './stream.js'
+import { createDigest } from './crypto.js'
+import { EventEmitter } from './events.js'
+
+async function* streamAsyncIterable(stream) {
+  const reader = stream.getReader();
```
Comment (heapwolf) [inline bootstrap.js]: can you run standard linter on this file? some of it is standard-ish, some of it not.

File: bootstrap.js [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+    }
+    const contentLength = +response.headers.get('Content-Length')
+    let receivedLength = 0
+    let prevProgress = 0
+    const fileData = new Uint8Array(contentLength)
+
+    for await (const chunk of streamAsyncIterable(response.body)) {
+      fileData.set(chunk, receivedLength)
+      receivedLength += chunk.length
+      const progress = (receivedLength / contentLength * 100) | 0
```
Comment (heapwolf) [inline bootstrap.js:117]: Are you sure you want to prevent slightly more granular progress with `| 0`? Visually its nice to have a more fluid progress.

**Review by heapwolf — APPROVED** [review]

**Pull request merged**